services/httpd/response_writer.go

```diff
@@ -25,19 +25,27 @@ type ResponseWriter interface {
 func NewResponseWriter(w http.ResponseWriter, r *http.Request) ResponseWriter {
 	pretty := r.URL.Query().Get("pretty") == "true"
 	rw := &responseWriter{ResponseWriter: w}
-	switch r.Header.Get("Accept") {
-	case "application/csv", "text/csv":
-		w.Header().Add("Content-Type", "text/csv")
-		rw.formatter = &csvFormatter{statementID: -1}
-	case "application/x-msgpack":
-		w.Header().Add("Content-Type", "application/x-msgpack")
-		rw.formatter = &msgpackFormatter{}
-	case "application/json":
-		fallthrough
-	default:
-		w.Header().Add("Content-Type", "application/json")
-		rw.formatter = &jsonFormatter{Pretty: pretty}
+
+	acceptHeaders := parseAccept(r.Header.Get("Accept"))
+	for _, accept := range acceptHeaders {
+		switch accept.Type + "/" + accept.SubType {
```

Comment:
It seems to me this code handles wildcards incorrectly. If I am understanding the spec correctly, a `*` would match any one of these. Is this correct? If it is, I have some recommendations for how we can change the code. If it isn't, then we can skip past that.

Reply:
Uhm, I agree. Makes sense.